modules/codemod/lib/v5/recategorizeButtons.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  API,
+  FileInfo,
+  Options,
+  StringLiteral,
+  JSXIdentifier,
+  JSXAttribute,
+  ASTPath,
+  JSXElement,
+} from 'jscodeshift';
+import {getImportRenameMap} from './getImportRenameMap';
+
+const updateJSXTag = (nodePath: ASTPath<JSXElement>, newTag: string) => {
+  (nodePath.value.openingElement.name as JSXIdentifier).name = newTag;
+  if (nodePath.value.closingElement) {
+    (nodePath.value.closingElement.name as JSXIdentifier).name = newTag;
+  }
+};
+
+export default function transformer(file: FileInfo, api: API, options: Options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  const requiredImportSpecifiers: string[] = [];
+
+  /**
+   * 1. Find button imports
+   */
+  const {containsCanvasImports} = getImportRenameMap(j, root, '@workday/canvas-kit-react/button');
+  if (!containsCanvasImports) {
+    return file.source;
+  }
+
+  /**
+   * 2. Find `Button`
+   *  - If it has variant="primary"
+   *    - Swap to `PrimaryButton` and remove variant prop
+   *    - Add `PrimaryButton` import if it doesn't exist.
+   *  - If not
+   *    - Swap to `SecondaryButton` and remove variant prop
+   *    - Add `SecondaryButton` import if it doesn't exist
+   */
+  root
+    .find(j.JSXElement, {openingElement: {name: {name: 'Button'}}})
```

Comment:
Note: These changes are based purely on the JSX element (therefore not informed by the import). This means there's a possibility of affecting the incorrect components. For example:

```
import {MyButton as Button} from 'my-project';
import {Button as CKRButton} from '@workday/canvas-kit-react';
```

It caught me on our `import {deprecated_Button as Button}` usage in the deprecated button stories.

It's going to add quite a bit of complexity to try to address this case. Do we think it's worth doing?

Reply:
There's a `getImportRenameMap` that both checks for Canvas-base imports and what the mappings are. You can do the following to check if the element was an imported `Button` from Canvas Kit:

```tsx

const {containsCanvasImports, importMap} = getImportRenameMap(
  j,
  root,
  '@workday/canvas-kit-react/button'
);

if (!containsCanvasImports) {
  return file.source
}

root
  .find(j.JSXElement, (node: JSXElement) => {
    // the following line might need additional `.type` checks
    if (node.openingElement.name.name === importMap.Button) {
      return true
    }
    return false
  })
```

If you want to even check for instances of `styled(Button)`, there's also a `styledMap` that detects styled renames, but only from CK imports.

Both are available from my PR: https://github.com/Workday/canvas-kit/pull/1028/files#diff-dc51f739fd52e024d8c2d94f0d87c81d9416fcd78c796376705c7af12fdaffa9R9

Reply:
@NicholasBoll After looking into it, it's still pretty complicated with that helper. It basically doubles the amount of logic in the codemod because I have to make different changes to both the element and the import depending on whether they use a namespaced import or not.

I can add it, but it's just a lot of complexity for something I'm not sure how many people are using.

Reply:
With the changes made in the Card PR, it isn't much extra code. Basically what I pasted minus the extra type checks. After the Card PR, all maps are any import names. `import.Button` will equal whatever the import name is. Default is `Button`. If they renamed it, it would be whatever the renamed value is. That was the reason for the update.